packages/apps/web-viewer-test/src/components/home/ViewerHome.tsx

```diff
@@ -83,20 +101,31 @@ const ViewerHome: React.FC = () => {
     if (urlParams.has("changesetId")) {
       setChangesetId(urlParams.get("changesetId") as string);
     }
-  }, []);
 
-  useEffect(() => {
-    let url = `viewer?iTwinId=${iTwinId}`;
+    if (urlParams.has("contextId")) {
+      setContextId(urlParams.get("contextId") as string);
+    }
 
-    if (iModelId) {
-      url = `${url}&iModelId=${iModelId}`;
+    if (urlParams.has("componentId")) {
+      setComponentId(urlParams.get("componentId") as string);
     }
 
-    if (changesetId) {
-      url = `${url}&changesetId=${changesetId}`;
+    if (urlParams.has("documentId")) {
+      setDocumentId(urlParams.get("documentId") as string);
     }
+  }, []);
+
+  useEffect(() => {
+    let url = `viewer?`;
+    allParams.forEach((param) => {
+      const sep = url.endsWith("?") ? "" : "&";
+      if (param) {
+        url += `${sep}${param}`;
+      }
+    });
```

Comment:
I think this section can be replaced with the following:

```ts
    const url = `viewer?${allParams.join("&")`
```

Reply:
I see now that my suggested change missed the `if (param)` line, although `filter` could be used to rectify that.

However, I don't understand how this new URL can work. It is lacking the `<name>=` part for each param. So, while before it would be something like `viewer?iTwinId=<blah>&iModelId=<yada>`, now you just have `viewer?<blah>&<yada>&&`.

Reply:
Yeah thanks. That's a miss on my part.